client/src/modules/list/index.jsx

```diff
@@ -67,41 +73,69 @@ class List extends Component {
         }
       });
 
-    joinRoom(Routes.LIST, cohortId, userId);
+    const roomConfig = {
+      subscribeMetaData: !isUnavailable,
+      resourceId: listId,
+      roomPrefix: Routes.LIST,
+      userId
+    };
+
+    joinRoom(roomConfig);
   }
 
   componentDidUpdate(previousProps) {
     const {
       list: previousList,
       match: {
         params: { id: previousListId }
-      }
+      },
+      members: previousMembers
     } = previousProps;
     const {
       currentUser: { id: userId },
       list,
       match: {
         params: { id: listId }
-      }
+      },
+      members
     } = this.props;
-
-    if (previousListId !== listId) {
-      leaveRoom(Routes.LIST, listId, userId);
+    const { isUnavailable } = this.state;
+    const hasListChanged = listId !== previousListId;
+    const roomConfig = {
+      subscribeMetaData: !isUnavailable,
+      resourceId: previousListId,
+      roomPrefix: Routes.LIST,
+      userId
+    };
+
+    if (hasListChanged) {
+      leaveRoom(roomConfig);
+      joinRoom({ ...roomConfig, resourceId: listId });
       this.fetchData();
     }
 
     if (previousList && list) {
       const {
-        name: previousName,
-        cohortName: previousCohortName
+        cohortName: previousCohortName,
+        name: previousName
       } = previousList;
       const { name, cohortName } = list;
       const updateBreadcrumbs =
         previousName !== name || previousCohortName !== cohortName;
+      const hasListBeenArchived = !previousList.isArchived && list.isArchived;
+      const hasListBeenRestored = previousList.isArchived && !list.isArchived;
+      const hasUserBeenRemoved = previousMembers[userId] && !members[userId];
+      const hasListBeenDeleted = !previousList.isDeleted && list.isDeleted;
 
       if (updateBreadcrumbs) {
         this.handleBreadcrumbs();
       }
+
+      if (hasListBeenArchived || hasUserBeenRemoved || hasListBeenDeleted) {
+        this.handleMakeListUnavailable();
+      } else if (hasListBeenRestored) {
```

Comment:
simplify to `else`

Reply:
It can not be simplified to else because when list is set as unavailable `didUpdate` is called (state was changed) and makes this list available again